osu.Game.Rulesets.Osu/Skinning/Default/DefaultFollowCircle.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Osu.Skinning.Default
+{
+    public class DefaultFollowCircle : CompositeDrawable
+    {
+        private DrawableSlider slider;
+
+        [BackgroundDependencyLoader]
+        private void load(DrawableHitObject drawableObject)
+        {
+            slider = (DrawableSlider)drawableObject;
+
+            RelativeSizeAxes = Axes.Both;
+            Alpha = 1f;
+
+            InternalChild = new CircularContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Masking = true,
+                BorderThickness = 5,
+                BorderColour = Color4.Orange,
+                Blending = BlendingParameters.Additive,
+                Child = new Box
+                {
+                    Colour = Color4.Orange,
+                    RelativeSizeAxes = Axes.Both,
+                    Alpha = 0.2f,
+                }
+            };
+
+            slider.Tracking.BindValueChanged(trackingChanged, true);
+            slider.ApplyCustomUpdateState += updateStateTransforms;
```

Comment:
These subscriptions are never removed, which I suspect is causing a delegate leak (they need to be removed on disposal/skin change). This goes for all four skinnables added here.

Generally binding to other bindables that a class does not own is frowned upon. The pattern to follow there is to create a local bindable and bind to the external one.